fab/fab/utils.py

```diff
@@ -132,6 +131,13 @@ def deploy_ref(self):
         # turn whatever `code_branch` is into a commit hash
         branch = repo.get_branch(self._code_branch)
         self._deploy_ref = branch.commit.sha
+
+        # Causes setup_release to fail fast if the right github permissions aren't set
+        repo.create_git_ref(
+            ref='refs/tags/' + '{}-{}-setup_release'.format(self.timestamp, self._environment),
+            sha=self._deploy_ref,
+        )
```

Comment:
If the error from the stack trace is too opaque, you could always catch the error you're expecting here and print out instructions for fixing it, and then exit. I'll leave that up to you to decide though.

Reply:
I'll do a follow up PR